data-prepper-plugins/grok-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepperTests.java

```diff
@@ -11,35 +11,93 @@
 
 package com.amazon.dataprepper.plugins.prepper.grok;
 
-import com.amazon.dataprepper.model.record.Record;
 
 import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.krakens.grok.api.Grok;
+import io.krakens.grok.api.GrokCompiler;
+import io.krakens.grok.api.Match;
+import com.amazon.dataprepper.model.record.Record;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.MockedStatic;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.Mock;
+import org.junit.jupiter.api.Test;
 
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.UUID;
 
+import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
 
 @ExtendWith(MockitoExtension.class)
 public class GrokPrepperTests {
 
-    PluginSetting pluginSetting;
+    private PluginSetting pluginSetting;
     private GrokPrepper grokPrepper;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<Map<String, Object>>() {};
+    private String messageInput;
+
+    @Mock
+    private GrokCompiler grokCompiler;
+
+    @Mock
+    private Match match;
+
+    @Mock
+    private Grok grok;
+
+    private final String PLUGIN_NAME = "grok";
+    private Map<String, Object> capture;
 
     @BeforeEach
     public void setup() {
-        pluginSetting = new PluginSetting(
-                "grok",
-                null
-                );
+        final Map<String, List<String>> matchConfig = new HashMap<>();
+        matchConfig.put("message", new ArrayList<String>() {{
```

Comment:
I see this occasionally as a short-hand. You are creating an anonymous inner class which is unnecessary. I think we should avoid this type of usage.

If this isn't modified later, just use `Collections.singletonList(...)`. If you modify it later, then do an `add` on its own line.